deps/rabbit/src/rabbit_amqp_reader.erl

```diff
@@ -136,7 +136,15 @@ system_code_change(Misc, _Module, _OldVsn, _Extra) ->
 server_properties() ->
     Props0 = rabbit_reader:server_properties(amqp_1_0),
     Props1 = [{{symbol, K}, {utf8, V}} || {K, longstr, V} <- Props0],
-    Props = [{{symbol, <<"node">>}, {utf8, atom_to_binary(node())}} | Props1],
+    Props = [
+             {{symbol, <<"node">>}, {utf8, atom_to_binary(node())}},
+             %% queue-prefix and topic-prefix are not part of the AMQP spec.
+             %% However, we include these properties because they are also returned by
+             %% ActiveMQ and Solace and understood by some client libs (e.g. ActiveMQ NMS.AMQP Client).
+             %% https://github.com/rabbitmq/rabbitmq-server/issues/12531
+             {{symbol, <<"queue-prefix">>}, {utf8, <<"/queues/">>}},
+             {{symbol, <<"topic-prefix">>}, {utf8, <<"/exchanges/amq.topic/">>}}
```

Comment:
@ansd - I've tested and for me it seems that there is a difference in behaviour if this is `/exchanges/amq.topic/` or `/topic/`. If I just subscribe to `test.*` topic the call succeeds with `/topic/`, but it fails if the prefix is `/exchanges/amq.topic/` - `Attach rejected: {unknown_destination,"/exchanges/amq.topic/test.*"}`

Reply:
Thanks for testing @alaendle.

Going forward RabbitMQ will support only address format v2: https://www.rabbitmq.com/docs/amqp#addresses
Therefore, including `/topic/` is not an option.

So far my understanding has been that the client libs use these prefixes only for **publishing** messages, i.e. as prefixes for **target** addresses. If these client libs also use these prefixes for **consuming** messages, i.e as **source** addresses, then RabbitMQ should advertise only the `/queues/` prefix.

Reply:
@alaendle where in the client libs to do you see these prefixes being used for **consuming** messages, i.e. as **source** address prefix?

Reply:
I changed the PR to include only the queue-prefix to be on the safe side.

Reply:
You are right! Because it started to work again if I remove the `s` from `exchanges` - means switching back to address format v1 😄. Unfortunately the translation need to work in both directions - so you're implementation is correct, but only supports v2 addresses. Only advertising the `queue-prefix` isn't a good solution - simple because of it's inconsistency - this would tell the user code to manage topic prefixes but not queue prefixes. So be honest I have to think about this a few minutes... if I use a queue for consumption (what I SHOULD do according to the documentation) everything works as expected; so two downsides of this PR - I could potentially break code that currently uses v1-prefixes like `/topic/` for consumption - because prefixes won't match and the client lib would build a address like `/exchanges/amq.topic//topic/test.*`; it would work if the client already follows v2 pattern on `amq.topic` - because if it is already named `/exchanges/amq.topic/test.cell` the prefix matches and the resulting address is identical. However code might break in another scenario - what if the client explicitly addresses another exchange like getting a topic for `/exchanges/amq.direct/test`? This would lead to: `Attach rejected: {bad_address_string,<<"/exchanges/amq.topic//exchanges/amq.direct/test">>`. I'm still pretty sure that it is worth to expose the properties and that this is exactly the behaviour all would like in 99% percent of all cases - however there are two caveats, a) code using v1-addresses could break, b) code using v2-addresses with exchanges other than `amq.topic` could break.

Reply:
> I changed the PR to include only the queue-prefix to be on the safe side.

You are too fast for me ⚡